src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7881,7 +7882,9 @@ private ForEachStatementSyntax ParseForEachStatement()
             var closeParen = this.EatToken(SyntaxKind.CloseParenToken);
             var statement = this.ParseEmbeddedStatement(true);
 
-            return _syntaxFactory.ForEachStatement(@foreach, openParen, type, name, deconstruction, @in, expression, closeParen, statement);
+            return (deconstruction == null)
+                ? (CommonForEachStatementSyntax)_syntaxFactory.ForEachStatement(@foreach, openParen, type, name, @in, expression, closeParen, statement)
```

Comment:
So that the ?: has a common type and will compile. Actually, the cast can be put on only one side, but I did it on both sides for symmetry.

---
In reply to: [73615221](https://github.com/dotnet/roslyn/pull/12917#discussion_r73615221) [](ancestors = 73615221)